### ISSUE TITLE
Bug WithRequestTimeout is not working properly 

### ISSUE THREAD

**Issue by nevesgustavo** [Flagsmith/flagsmith-go-client#43]
I noticed that the method `WithRequestTimeout` is not working properly because it's receiving a `time.Duration` and in the `NewFlagsmith` method it's multiplied for `time.Second`. Please see the details below.

1 - The timeout property is a time.Duration [Timeout](https://github.com/Flagsmith/flagsmith-go-client/blob/main/config.go#L19)

2 - In the [NewClient](https://github.com/Flagsmith/flagsmith-go-client/blob/main/client.go#L47) the timeout is multiplied for `time.Second` `c.client.SetTimeout(time.Second * c.config.timeout)`

3 - As you can see in the images below after the multiplication the timeout prop is wrong
![time duration](https://user-images.githubusercontent.com/10534124/191143705-18ccebcd-4851-444b-9518-b1cb599d6d1d.png)
![wrong value](https://user-images.githubusercontent.com/10534124/191143716-d7d1c47a-e6db-47ea-8342-dc8e06c31dbb.png)

**Solution suggestion**  

1 - Change this line [NewClient](https://github.com/Flagsmith/flagsmith-go-client/blob/main/client.go#L47) to `c.client.SetTimeout(c.config.timeout)` instead of `c.client.SetTimeout(time.Second * c.config.timeout)`

2 - Or change this property to int instead of `time.Duration` [Timeout](https://github.com/Flagsmith/flagsmith-go-client/blob/main/config.go#L19)

**Comment by gagantrivedi** [comment]
@nevesgustavo Thanks for the detailed issue description.  I will release the patch in a couple of hours

**Comment by nevesgustavo** [comment]
@gagantrivedi thanks for the quick reply. Additionally I think that this line [Timeout](https://github.com/Flagsmith/flagsmith-go-client/blob/main/client.go#L47) needs to be moved after this for [Setting options](https://github.com/Flagsmith/flagsmith-go-client/blob/main/client.go#L49) because in the opt(c) is setting the `c.config.timeout` so I think that the SetTimeout needs to run after this.

Thanks again

**Comment by gagantrivedi** [comment]
> 

Ah, on the placement is correct (else it will override the value set by the user explicitly ) I think, but the option method is wrong it should be 

```
func WithRequestTimeout(timeout time.Duration) Option {
	return func(c *Client) {
		c.client.SetTimeout(timeout)
	}
}
```

**Comment by nevesgustavo** [comment]
It looks great, I think will work properly.

Thank you @gagantrivedi

**Comment by nevesgustavo** [comment]
Hi @gagantrivedi, how's it going?

Could you please tell me when it'll be available? and also if you've added the fix about the WithRequestTimeout that we've talked before.

Thanks a lot :)

```go
func WithRequestTimeout(timeout time.Duration) Option {
	return func(c *Client) {
		c.client.SetTimeout(timeout)
	}
}
```

**Comment by gagantrivedi** [comment]
Hi, yes, it's live on v2.0.1. 
just do `go get github.com/Flagsmith/flagsmith-go-client/v2`

**Comment by gagantrivedi** [comment]
Fixed in release 2.0.1
